### PR TITLE
Implement a Redux-store driven approach for the Permission and Role Create/Edit/Delete modals

### PR DESCRIPTION
**Pull request by chaidas**
@jmike as discussed, this PR has the redux-store driven approach for the Create / Edit and Delete modals for both Permissions and Roles.

I opted to split the generic `modalStore` into two separate stores (`roleModalsStore.js` as well as `permissionModalsStore.js`) to make things clearer. The store structure is quite clear with this approach:

image

These modals reside in `/app/modals` instead of the "appropriate" place because they can easily be reused from _any_ page in the admin ui (as a matter of fact the Create and Edit modals for both Permissions and Roles are used from within the `OrgEdit(Permission|Role).js` pages).

### CODE REVIEW

File: admin_ui/app/modals/PermissionDeleteModal.js [hunk 1]
```diff
@@ -25,7 +23,7 @@ const PermissionDeleteModal = () => {
     dispatch(deletePermission(permission));
   }, [dispatch, permission]);
 
-  if (!permission.id) {
+  if (displayedModal !== 'PERMISSION_DELETE' || !permission.id) {
```
Comment (jmike) [inline admin_ui/app/modals/PermissionDeleteModal.js]: Q: Do we need this second check? When will `displayModal` be "PERMISSION_DELETE" and `permission.id` be null? Can you give an example?

File: admin_ui/app/modals/permissionModalsStore.js [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+};
+
+// ACTIONS
+
+// ****************************************************************************
+// Permission CREATE
+// ****************************************************************************
+export const openPermissionCreateModal = createAction(
+  'PERMISSION_CREATE_MODAL_OPEN',
+  (onSubmit = noop, onCancel = noop) => {
```
Comment (jmike) [inline admin_ui/app/modals/permissionModalsStore.js]: Arguments as function params make it harder to refactor in the future (e.g. add yet another argument) - whereas passing arguments as object props makes the API more flexible. I prefer the latter, although I have no strong opinion.

File: admin_ui/app/modals/PermissionDeleteModal.js (hunk 1 above)
Comment (chaidas) [reply admin_ui/app/modals/PermissionDeleteModal.js]: Hmmm nope we don't. This was me just coding super-defensively :-)

File: admin_ui/app/modals/permissionModalsStore.js (hunk 2 above)
Comment (chaidas) [reply admin_ui/app/modals/permissionModalsStore.js]: I think we blindly followed the structure for a `payloadCreator` method as specified here: https://redux-actions.js.org/api/createaction. Or am I missing something on this comment?

**Pull request merged**